swiffer-api/src/main/java/com/solambda/swiffer/api/internal/decisions/DecisionTaskContextImpl.java

```diff
@@ -59,10 +66,31 @@ public VersionedName workflowType() {
 		return new VersionedName(workflowType.getName(), workflowType.getVersion());
 	}
 
+    @Override
+    public boolean hasMarker(String markerName) {
+        return decisionTask.getEvents().stream().anyMatch(isMarkerRecordedEvent(markerName));
+    }
+
+	@Override
+    public <T> Optional<T> getMarkerDetails(String markerName, Class<T> type) {
+        return decisionTask.getEvents().stream()
+                           .filter(isMarkerRecordedEvent(markerName))
+                           .findFirst()
+                           .map(deserialize(type));
+	}
+
 	@Override
 	public String toString() {
 		return "DecisionTaskContextImpl [domain=" + this.domain
 				+ ", workflowType=" + workflowType() + "]";
 	}
 
+    private static Predicate<HistoryEvent> isMarkerRecordedEvent(String markerName) {
+        return event -> event.getMarkerRecordedEventAttributes() != null
```

Comment:
This method is designed to be used with markers without details. 
Replacing it with event.details() wouldn't return correct result for markers without details.